docs/Plugins.md

```diff
@@ -0,0 +1,75 @@
+<h1 align="center">Fastify</h1>
+
+## Plugins
+Fastify allows the user to extend its functionalities with plugins.
+A plugin can be a set of routes, a server decorator or whatever. The API that you will need to use one or more plugins, is `register`.  
+You already see in the [getting started](https://github.com/fastify/fastify/blob/master/docs/Getting-Started.md#register) section how use this API, is pretty straightforward.
+```js
+fastify.register([
+  require('./another-route'),
+  require('./yet-another-route')
+], opts, function (err) {
+  if (err) throw err
+})
+```
+
+By default, `register` creates a *new scope*, this means that everything you do to the current Fastify instance is not reflected to the current context ancestors, but only to its sons. This feature allows us to achieve plugin *encapsulation* and *inheritance*.
+
+
+### Use a plugin
+Use a plugin is very easy, you just need to require it and pass it as parameter to the `register` API.
+You can pass a single plugin or an array of plugins if all of them can share the same options object/callback.
+
+Sometimes you don't want to use the plugin *encapsulation*, in that cases you have two ways to tell Fastify to skip the creation of a new scope:
+- Use the [`fastify-plugin`](https://github.com/fastify/fastify-plugin) module.
+- Use the `'skip-override'` hidden property
+
+We recommend to use the `fastify-plugin` module.
+```js
+const fp = require('fastify-plugin')
+
+module.exports = fp(function (fastify, opts, next) {
+  // your plugin code
+  next()
+})
+```
```

Comment:
this information is repeated in the next section, maybe we can just link down to the relative sections in the list.